homeassistant/components/tplink/light.py

```diff
@@ -294,36 +297,38 @@ def supported_features(self) -> int:
         """Flag supported features."""
         return self._light_features.supported_features
 
-    def _get_valid_temperature_range(self) -> tuple[int, int]:
+    @staticmethod
+    def _get_valid_temperature_range(sys_info: dict[str, Any]) -> tuple[int, int]:
```

Comment:
It may be good to break out all the device api methods to separate class that we instantiate and store on the entity for use. Mixing the entity api with the device api makes it hard to see what's going on.